norduser/process/norduser_process_manager.go

```diff
@@ -0,0 +1,93 @@
+package process
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"sync"
+
+	childprocess "github.com/NordSecurity/nordvpn-linux/child_process"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/NordSecurity/nordvpn-linux/norduser/pb"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func NorduserURL(uid uint32) string {
+	return fmt.Sprintf("%s://%s", internal.Proto, internal.GetNorduserSocketSnap(uid))
+}
+
+type NorduserProcessClient struct {
+	mu  sync.Mutex
+	uid uint32
+}
+
+func NewNorduserProcessClient(uid uint32) *NorduserProcessClient {
+	return &NorduserProcessClient{
+		uid: uid,
+	}
+}
+
+func getNorduserClient(uid uint32) (pb.NorduserClient, *grpc.ClientConn, error) {
```

Comment:
lot of code from here is the same as is [fileshare_process_manager.go](https://github.com/NordSecurity/nordvpn-linux/pull/321/files#diff-8d942a082510951ed70f6e44609977dcd7bfbe7661628b049366afc8e52a7bda)
would it be possible in the future to mode them in one place?

Reply:
Generally most of the code in fileshare/norduser process managers is trivial and making it generic would require some generics or factory convolution(it all comes down to the fact that grpc clients are of a different type). So I just chose to duplicate it, because it was relatively low cost. But there is no problem to change it if preference is to have it commonized.